lib/install/management/upgrade.go

```diff
@@ -69,17 +69,35 @@ func (d *Dispatcher) Upgrade(vch *vm.VirtualMachine, conf *config.VirtualContain
 	if err = d.uploadImages(settings.ImageFiles); err != nil {
 		return errors.Errorf("Uploading images failed with %s. Exiting...", err)
 	}
+	var deleteImage bool
+	defer func() {
```

Comment:
it seems like defer just complicates the code. I would rather call an appropriate function right before return, since it is clear where it should happens.